packages/hoppscotch-common/src/pages/profile.vue

```diff
@@ -277,9 +278,20 @@ const updatingEmailAddress = ref(false)
 watchEffect(() => (emailAddress.value = currentUser.value?.email || ""))
 
 const updateEmailAddress = () => {
+  const inputEmailAddress = emailAddress.value.trim()
+  if (!inputEmailAddress) {
+    toast.error(`${t("error.empty_email_address")}`)
+    return
+  }
+
+  if (currentUser.value?.email === inputEmailAddress) {
+    toast.error(`${t("error.same_email_address")}`)
+    return
+  }
+
   updatingEmailAddress.value = true
   platform.auth
-    .setEmailAddress(emailAddress.value as string)
+    .setEmailAddress(inputEmailAddress as string)
```

Comment:
On another note, the Email input field can be marked read-only for self-hosted instances since the underlying action isn't in place.

Note: This is to be revisited by the core team after relevant discussions.

https://github.com/hoppscotch/hoppscotch/blob/80b63545f14b43067ba290ab6bf15e3d8438f966/packages/hoppscotch-selfhost-web/src/platform/auth/web/index.ts#L319-L322